docs/reference/analysis/tokenfilters/delimited-payload-tokenfilter.asciidoc

```diff
@@ -1,21 +1,331 @@
 [[analysis-delimited-payload-tokenfilter]]
-=== Delimited Payload Token Filter
-
-Named `delimited_payload`. Splits tokens into tokens and payload whenever a delimiter character is found.
+=== Delimited payload token filter
+++++
+<titleabbrev>Delimited payload</titleabbrev>
+++++
 
 [WARNING]
-============================================
+====
+The older name `delimited_payload_filter` is deprecated and should not be used
+with new indices. Use `delimited_payload` instead.
+====
+
+Separates a token stream into tokens and payloads based on a specified
+delimiter.
+
+For example, you can use the `delimited_payload` filter with a `|` delimiter to
+split `the|1 quick|2 fox|3` into the tokens `the`, `quick`, and `fox`
+with respective payloads of `1`, `2`, and `3`.
+
+This filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/payloads/DelimitedPayloadTokenFilter.html[DelimitedPayloadTokenFilter].
+
+[NOTE]
+.Payloads
+====
+A payload is user-defined binary data associated with a token position and
+stored as base64-encoded bytes. Payloads are often used with the
+<<query-dsl-script-score-query,`script_score`>> query to calculate custom scores
```

Comment:
I guess would be nice to do this some time, but currently there is no way for `script_score` to access payloads.  I don't know any other ES query that can deal with payloads either. The only way to access them is through _termvectors API.

Reply:
Thank you for catching this. I've updated this note to state that you can view stored payloads using the term vectors API API.